test/gurka/test_search_filter.cc

```diff
@@ -0,0 +1,221 @@
+#include "gurka.h"
+#include <boost/format.hpp>
+#include <gtest/gtest.h>
+
+using namespace valhalla;
+
+/*************************************************************/
+class HeadingFilter : public ::testing::Test {
+protected:
+  static gurka::map map;
+
+  static void SetUpTestSuite() {
+    constexpr double gridsize = 100;
+
+    const std::string ascii_map = R"(
+    B----2----C
+    |         |
+    |1        |
+    A---------D
+         )";
+    const auto layout = gurka::detail::map_to_coordinates(ascii_map, gridsize);
+    const gurka::ways ways = {{"AB", {{"highway", "primary"}}},
+                              {"BC", {{"highway", "primary"}}},
+                              {"CD", {{"highway", "primary"}}},
+                              {"AD", {{"highway", "primary"}}}};
+    map = gurka::buildtiles(layout, ways, {}, {}, "test/data/search_filter_heading");
```

Comment:
Does `buildtiles` do nothing if invoked again with the same tileset? If not, this pattern may result in several runs of the tile baking process. (Maybe it's lightning quick given the small graph).

Reply:
It'll rebuild tiles on every run, and can be a bit slow (on the order of 5-10 seconds per build). We should try to group test maps together to minimize build times, so I simplified the tests to that end in https://github.com/valhalla/valhalla/pull/2289/commits/a73ed027b864c7b5a1a040c204e5d13e3782fd20.